packages/dapp/src/helpers/listingEvents.ts

```diff
@@ -1,10 +1,20 @@
 import { Dispatch } from "redux";
-import { getTCR } from "./civilInstance";
+import { getTCR, getCivil } from "./civilInstance";
 import { addListing } from "../actionCreators/listings";
+import { addNewsroom } from "../actionCreators/newsrooms";
+import { EthAddress } from "@joincivil/core";
 
 export async function initializeSubscriptions(dispatch: Dispatch): Promise<void> {
   const tcr = getTCR();
   await tcr.allApplicationsEver().forEach(async listing => {
     dispatch(addListing(listing));
+    await getNewsroom(dispatch, listing.address);
```

Comment:
the timing of these may be bad in some situations, I would do both dispatches after the promise.

Reply:
just moving `await getNewsroom...` above `dispatch(addListing...` accomplishes that, right?

Reply:
I think so, test it out